### ISSUE TITLE
PointNet++ Support

### ISSUE THREAD

**Issue by yang-zhifei** [ChrisWu1997/DeepCAD#2]
As you described in the applications

using pointnet2 to encoder the point cloud to the latent code?

Can you release the details of it?

I want to re-implement this experiment. 

THX

**Comment by ChrisWu1997** [comment]
Hi, 

I just upload `pc2cad.py` file. Please take a look. In order to run it, you need to convert all CAD models to point clouds first, and remember to install https://github.com/erikwijmans/Pointnet2_PyTorch.

Though we train it on all of our data, this straight-forward pc2cad method still doesn't generalize well (see Table 6 at supplementary and compare the numbers to Table 2 autoencoding results).

**Comment by yang-zhifei** [comment]
Hello,

thanks for your reply, I have used the generated point clond and `pc2cad.py` for training for 20 hours, but the loss has not dropped.

 I can't re-implement this experiment.

Do you have any suggestions?

![2021-09-28 18-49-06屏幕截图](https://user-images.githubusercontent.com/55872024/135073566-2e5bec6d-deb5-47f7-8696-ee796e258b74.png)


![2021-09-28 18-32-19屏幕截图](https://user-images.githubusercontent.com/55872024/135072370-616c0762-fe56-42e9-9042-308ba50dc2f5.png)

**Comment by ChrisWu1997** [comment]
My training curve goes like this:
![image](https://user-images.githubusercontent.com/32172140/135118091-5f495640-1810-42b8-a426-b9fed246e631.png)

The loss goes below 0.07. Maybe you can check the scale of your point clouds? It should be of the same size as CAD model. In other words, you should normalize the CAD model before converting to point clouds.

**Comment by yang-zhifei** [comment]
@ChrisWu1997 

Here is my solution to this problem


1. Transform the CAD Sequence commands to Point Cloud, Use 

```bash
cd dataset
python jsoin2pc.py 
```

normalize the CAD model means that?
https://github.com/ChrisWu1997/DeepCAD/blob/master/dataset/json2pc.py#L46-L53

2. Train on the generated point cloud data set, Use 
```bash
python pc2cad.py --ae_exp_name newDeepCAD --exp_name Point2Enc --ae_ckpt 1000 -g 1,2,3
```

**Comment by ChrisWu1997** [comment]
Hi @yang-zhifei,

I retried the code:
```shell
cd dataset
python json2pc.py
cd ..
python pc2cad.py --exp_name newDeepCAD --ae_ckpt 1000 -g 3 --pc_root {path-to-pc-data}
```

It works fine for me, as the loss quickly goes down below 0.07.
![image](https://user-images.githubusercontent.com/32172140/135517177-22bcdca0-bb8f-4d61-ab13-8369de1c92fb.png)

Are you using the pretrained AE model? If not, maybe you can try the pretrained AE model first. In addition, I see you are using multiple GPUs. Batch normalization has a different behavior by default for parallel training. Not sure if that's the cause of the problem, but maybe you can try using single GPU with smaller batch size.

**Comment by yang-zhifei** [comment]
Hi @ChrisWu1997,

Thank you very much for your help, my problem has been solved by running your code, it is a problem with my code.

![2021-10-03 23-04-39屏幕截图](https://user-images.githubusercontent.com/55872024/135760249-3ec3ff3b-19c0-4ae1-a7c6-c4df57422799.png)
